scripts/rewrite-declation-statements-for-typescript.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * @fileoverview Rewrites JS to match relative path format. That means
+ *  * Rewrite import FooConstants from '@material/foo/constants' to import FooConstants from './../foo/index'
+ *  * Rewrite import {FooFoundation} from '@material/foo' to import {FooFoundation} from './mdc-foo/index'
+ *
+ *
+ * This script rewrites import statements such that:
+ *
+ * ```js
+ * import [<SPECIFIERS> from] '@material/$PKG[/files...]';
+ * ```
+ * becomes
+ * ```js
+ * import [<SPECIFIERS> from] '$PKG.<RESOLVED_FILE_NAMESPACE>';
+ * ```
+ * The RESOLVED_FILE_NAMESPACE is a namespace matching the directory structure.
+ *
+ * This script also handles third-party dependencies, e.g.
+ *
+ * ```js
+ * import {thing1, thing2} from 'third-party-lib';
+ * ```
+ *
+ * becomes
+ *
+ * ```js
+ * import {thing1, thing2} from 'mdc.thirdparty.thirdPartyLib';
+ * ```
+ *
+ * and
+ *
+ * ```js
+ * import someDefaultExport from 'third-party-lib';
+ * ```
+ *
+ * becomes
+ *
+ * ```js
+ * import {default: someDefaultExport} from 'mdc.thirdparty.thirdPartyLib'
+ * ```
+ *
+ * Note that for third-party modules, they must be defined in closure_externs.js. See that file for more info.
```

Comment:
This is talking about closure_externs. Is this stale?

Reply:
removed